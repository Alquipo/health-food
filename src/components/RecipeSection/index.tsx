import * as S from './styles'

import Container from '../Container'
import RecipeCard from '../RecipeCard'

export default function RecipeSection() {
  return (
    <S.Wrapper>
      <Container>
        <S.Content>
          <S.Header>
            <S.Title>Our Best Recipes</S.Title>

            <S.Text>
              Far far away, behind the word mountains, far from the countries
              Vokalia and Consonantia, there live the blind texts.
            </S.Text>
          </S.Header>

          <S.Recipes>
            <RecipeCard
              image="/img/comida_1.svg"
              title="Broccoli Saladwith Bacon"
            />
            <RecipeCard
              image="/img/comida_2.svg"
              title="Classic Beef Burgers"
            />
            <RecipeCard
              image="/img/comida_3.svg"
              title="Classic Potato Salad"
            />
            <RecipeCard
              image="/img/comida_4.svg"
              title="Cherry Cobbler on the Grill"
            />
          </S.Recipes>
        </S.Content>
      </Container>
    </S.Wrapper>
  )
}
