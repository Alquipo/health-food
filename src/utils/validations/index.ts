import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
