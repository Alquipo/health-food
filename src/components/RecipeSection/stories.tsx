import { Story, Meta } from '@storybook/react'
import RecipeSection from '.'

export default {
  title: 'RecipeSection',
  component: RecipeSection
} as Meta

export const Default: Story = () => <RecipeSection />
