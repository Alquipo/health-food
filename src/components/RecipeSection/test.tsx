import { render, screen } from 'utils/test-utils'

import RecipeSection from '.'

jest.mock('components/RecipeCard', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock RecipeCard"></div>
    }
  }
})

describe('<RecipeSection/>', () => {
  it('should render a service section correctly', () => {
    const { container } = render(<RecipeSection />)

    expect(
      screen.getByRole('heading', {
        name: /Our Best Recipes/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        /Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts./i
      )
    ).toBeInTheDocument()

    expect(screen.getAllByTestId('Mock RecipeCard')).toHaveLength(4)
    expect(container.firstChild).toMatchSnapshot()
  })
})
