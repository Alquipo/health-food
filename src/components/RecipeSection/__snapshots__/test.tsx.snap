// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeSection/> should render a service section correctly 1`] = `
.c0 {
  background: #FAFAFC;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5.6rem 0 4.0rem;
}

.c3 {
  width: 100%;
  max-width: 51rem;
}

.c4 {
  font-size: 3.6rem;
  color: #1D164D;
  text-align: center;
}

.c5 {
  margin-top: 2.4rem;
  color: #9E9BAF;
  text-align: center;
  line-height: 1.4;
}

.c6 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 100%;
  gap: 3.2rem;
  margin-top: 5.6rem;
}

.c1 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 1.5rem;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Our Best Recipes
        </h2>
        <p
          class="c5"
        >
          Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
        </p>
      </div>
      <div
        class="c6"
      >
        <div
          data-testid="Mock RecipeCard"
        />
        <div
          data-testid="Mock RecipeCard"
        />
        <div
          data-testid="Mock RecipeCard"
        />
        <div
          data-testid="Mock RecipeCard"
        />
      </div>
    </div>
  </div>
</section>
`;
